tests/test_vocabulary.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+
+from jschon import URIError, JSONSchema
+
+@pytest.mark.parametrize('draft', ('2019-09', '2020-12', 'next'))
+def test_non_normalized_id_allowed(draft):
+    schema_data = {
+        '$schema': f'https://json-schema.org/draft/{draft}/schema',
+        '$id': 'http://localhost:1234/foo/bar/../baz',
+    }
+    schema = JSONSchema(schema_data)
+    assert str(schema['$id'].value) == schema_data['$id']
+
+
+@pytest.mark.parametrize('draft', ('2019-09', '2020-12', 'next'))
+def test_nonempty_fragment_id_disallowed(draft):
+    schema_data = {
+        '$schema': f'https://json-schema.org/draft/{draft}/schema',
+        '$id': '#foo',
+    }
+    match = 'a fragment' if draft == 'next' else 'a non-empty fragment'
```

Comment:
Just to make it easier to understand what's going on here at first glance, could you rename `match` to `error`, and make the match strings `has a fragment` and `has a non-empty fragment` respectively.

Reply:
I think it would be better to reverse the condition, i.e.
`error = 'has a non-empty fragment' if draft in ('2019-09', '2020-12') else ...`
to avoid having to update the test function body when future versions are published.